arangod/Cluster/ClusterInfo.cpp

```diff
@@ -462,71 +466,97 @@ void ClusterInfo::loadPlan() {
 
-  if (result.successful()) {
-    VPackSlice slice = result.slice()[0].get(
-        std::vector<std::string>({AgencyCommManager::path(), "Plan"}));
-    auto planBuilder = std::make_shared<VPackBuilder>();
-    planBuilder->add(slice);
+  if (!result.successful()) {
+    LOG_TOPIC(DEBUG, Logger::CLUSTER)
+      << "Error while loading " << prefixPlan
+      << " httpCode: " << result.httpCode() << " errorCode: " << result.errorCode()
+      << " errorMessage: " << result.errorMessage() << " body: " << result.body();
 
-    VPackSlice planSlice = planBuilder->slice();
+    return;
+  }
 
-    if (planSlice.isObject()) {
-      uint64_t newPlanVersion = 0;
-      VPackSlice planVersionSlice = planSlice.get("Version");
-      if (planVersionSlice.isNumber()) {
-        try {
-          newPlanVersion = planVersionSlice.getNumber<uint64_t>();
-        } catch (...) {
-        }
-      }
-      LOG_TOPIC(TRACE, Logger::CLUSTER) << "loadPlan: newPlanVersion=" << newPlanVersion;
-      if (newPlanVersion == 0) {
-        LOG_TOPIC(WARN, Logger::CLUSTER) << "Attention: /arango/Plan/Version "
-                                            "in the agency is not set or not "
-                                            "a positive number.";
-      }
-      {
-        READ_LOCKER(guard, _planProt.lock);
-        if (_planProt.isValid && newPlanVersion <= _planVersion) {
-          LOG_TOPIC(DEBUG, Logger::CLUSTER)
-              << "We already know this or a later version, do not update. "
-              << "newPlanVersion=" << newPlanVersion << " _planVersion=" << _planVersion;
-          return;
-        }
-      }
-      decltype(_plannedDatabases) newDatabases;
-      decltype(_plannedCollections) newCollections;  // map<string /*database id*/
-                                                     //    ,map<string /*collection id*/
-                                                     //        ,shared_ptr<LogicalCollection>
-                                                     //        >
-                                                     //    >
-      decltype(_shards) newShards;
-      decltype(_shardServers) newShardServers;
-      decltype(_shardKeys) newShardKeys;
-
-      bool swapDatabases = false;
-      bool swapCollections = false;
-      bool swapViews = false;
-
-      VPackSlice databasesSlice;
-      databasesSlice = planSlice.get("Databases");
-      if (databasesSlice.isObject()) {
-        std::string name;
-        for (auto const& database : VPackObjectIterator(databasesSlice)) {
-          try {
-            name = database.key.copyString();
-          } catch (arangodb::velocypack::Exception const& e) {
-            LOG_TOPIC(ERR, Logger::AGENCY)
-                << "Failed to get database name from json, error '" << e.what()
-                << "'. VelocyPack: " << database.key.toJson();
-            throw;
-          }
+  auto slice = result.slice()[0].get( // get slice
```

Comment:
should we check if `result().slice()` is array?

Reply:
done